recipe/meta.yaml

```
@@ -1,58 +1,74 @@
-{% set version = '3.1.4' %}
-
-{% if win %}
-  # msmpi is the only MPI variant for Windows
-  {% set mpi ='msmpi' %}
-{% else %}
-  {% set mpi = mpi or 'mpich' %}
-{% endif %}
+{% set version = '4.0.3' %}
+{% set name = "mpi4py" %}
 
 package:
-  name: mpi4py
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  fn: mpi4py-{{ version }}.tar.gz
-  url: https://github.com/mpi4py/mpi4py/releases/download/{{ version }}/mpi4py-{{ version }}.tar.gz
-  sha256: 17858f2ebc623220d0120d1fa8d428d033dde749c4bc35b33d81a66ad7f93480
+  url: https://github.com/{{ name }}/{{ name }}/releases/download/{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: de2710d73e25e115865a3ab63d34a54b2d8608b724f761c567b6ad58dd475609
 
 build:
   number: 0
-  script: 
-    - {{ PYTHON }} conf/cythonize.py
-    - {{ PYTHON }} -m pip install --no-deps . -vv
-  # s390x is missing a mpi variant (no openmpi or mpich)
-  skip: True  # [py<36 or s390x]
-
+  script:
+    - {{ PYTHON }} -m pip install --no-deps --no-build-isolation . -vv
 requirements:
   build:
     - {{ compiler('c') }}
   host:
     - python
     - pip
-    - cython
-    - setuptools
     - wheel
-    - {{ mpi }}
+    - setuptools >=42
+    - cython
+    - mpich  # [unix]
+    - msmpi  # [win]
   run:
     - python
-    - {{ mpi }}
+    - mpich  # [unix]
+    - msmpi  # [win]
+
+# ModuleNotFoundError: No module named 'test_util_pool'
+{% set tests_to_skip = "TestProcessPool" %}
 
 test:
+  source_files:
+    - test
+  imports:
+    - mpi4py
+    - mpi4py.MPI
+    - mpi4py.futures
+    - mpi4py.util.dtlib
+    - mpi4py.util.pkl5
+    - mpi4py.util.sync
+    - mpi4py.util.pool
+  commands:
+    - pip check
+    - python -c "from importlib.metadata import version; assert(version('{{ name }}')=='{{ version }}')"
+    # On Linux/Windows full testscope hanging without any issue. Re-check during version update.
+    - pytest -v -k "not {{ tests_to_skip }}" test #  [osx]
+    - pytest -v test/test_toplevel.py test/test_environ.py test/test_package.py test/test_pack.py test/test_attributes.py  #  [win or linux]
```

```suggestion
    - pytest -v test  #  [win or linux]
```
And why is OSX disabled?

---

https://github.com/mpi4py/mpi4py/tree/master/test
take a look

---

and enable tests for osx taget

---

Find where you have a problem and disable only that test.